art/attacks/evasion/fast_gradient.py

```diff
@@ -487,3 +490,36 @@ def _compute(
                     x_adv[batch_index_1:batch_index_2] = x_init[batch_index_1:batch_index_2] + perturbation
 
         return x_adv
+
+    @staticmethod
+    def _get_mask(x: np.ndarray, **kwargs) -> np.ndarray:
```

Comment:
This method is overwriting `_get_mask` defined in lines 194ff. Since the method defined here is more comprehensive, perhaps the above one should be deleted? Also, should this method be called in l.229-232 to fetch the mask from `kwargs` and perform the basic shape consistency check?

Reply:
Similarly, do we want to call this method in `ProjectedGradientDescentNumpy` l. 249-253, `ProjectedGradientDescentPyTorch` l. 123-125, `ProjectedGradientDescentTensorFlowV2`, l. 122-124?